src/compiler/emitter.js

```diff
@@ -267,7 +267,7 @@ var ts;
                 var sourceMapNameIndexMap = {};
                 var sourceMapNameIndices = [];
                 function getSourceMapNameIndex() {
-                    return sourceMapNameIndices.length ? sourceMapNameIndices[sourceMapNameIndices.length - 1] : -1;
+                    return sourceMapNameIndices.length ? lastOrUndefined(sourceMapNameIndices) : -1;

```

Comment:
lastOrUndefined is for calling on arrays that might be empty.  You should never actually get undefined back in these cases, and we'd want to throw if you did.  We should just have an 'last' functoin that only works on non-empty arrays, and throws otherwise.